### PR TITLE
ResponseStatusExceptionFilter: Fix unwrap of JSON message

### LINKED ISSUE #1222: NotFoundException message is empty since 3.1.3
Before version 3.1.3 removing an unexisting container was throwing an NotFoundException with a message, example: `{"message":"No such container: abcdefgh1234567890"}`.

Since version 3.1.3, the message of the NotFoundException is null.

This issue is preventing to upgrade from version 3.1.2 to 3.1.3 in bmuschko/gradle-docker-plugin, see failing tests of PR https://github.com/bmuschko/gradle-docker-plugin/pull/840

This is the commit causing the issue: https://github.com/docker-java/docker-java/commit/32b4db656a428fa6cd1f5b64e124954e9d187130

### PR DESCRIPTION
**Pull request by hugares**
Reading the entity stream a second time without resetting it was causing
the ObjectMapper to return null then a NullPointerExceptions when
calling get('message') method on it.

Fix the issue by creating the ObjectMapper from the String
representation of the entity stream instead of resetting the stream to
read it a second time. Also check to make sure the returned JsonNode is
not null and that the content is text.

Closes #1222

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/1223)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java [hunk 1]
```diff
@@ -80,9 +80,12 @@ private String getBodyAsMessage(ClientResponseContext responseContext) {
 
                 if (MediaType.APPLICATION_JSON_TYPE.equals(mediaType)) {
                     ObjectMapper mapper = new ObjectMapper();
-                    JsonNode node = mapper.readTree(entityStream).get("message");
+                    JsonNode node = mapper.readTree(message);
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: maybe surround with try? Parsing here is optional and shouldn't fail response exceptions

Comment (hugares) [reply src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: Done

File: src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java [hunk 2]
```diff
@@ -79,13 +79,18 @@ private String getBodyAsMessage(ClientResponseContext responseContext) {
...
+                    try {
+                        JsonNode node = new ObjectMapper().readTree(message);
+                        if (node != null) {
+                            JsonNode messageNode = node.get("message");
+                            if (messageNode != null && messageNode.isTextual()) {
+                                message = messageNode.textValue();
+                            }
+                        }
+                    } catch (IOException ignored) {
+                        //ignore parsing errors and return the message as is
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: please log to error logger, that may help to identify issue and fix in future

Comment (hugares) [reply src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: Done, I also logged the stack trace. let me know if you prefer to log only the error message

File: src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java [hunk 3]
```diff
@@ -31,6 +33,8 @@
  */
 public class ResponseStatusExceptionFilter implements ClientResponseFilter {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResponseStatusExceptionFilter.class.getName());
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: without `.getName()` just class is enough

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: `LOG`

Comment (hugares) [reply src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: Done

Comment (hugares) [reply src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: Done

Comment (hugares) [reply src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: FYI, there is 59 places using `LOG` vs 147 for `LOGGER`, should we consider changing the others for consistency (In a separate PR of course)?

Comment (KostyaSha) [reply src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: to minimise new lines i'm using shorter variant. No need to PR, not critical.

Comment (hugares) [reply src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java]: OK

**Pull request merged**